### PR TITLE
feat(library): add Zscaler AI Guard integration

### PR DESCRIPTION
**Pull request by willguibr**
## Summary

Adds a new community library integration for [Zscaler AI Guard](https://help.zscaler.com/ai-guard), providing input and output moderation rails that scan prompts and LLM responses for security threats using the Zscaler AI Guard DAS API via the [zscaler-sdk-python](https://pypi.org/project/zscaler-sdk-python/) SDK.

### Key features

- **Input & output rails** — Colang 1.0 (`flows.v1.co`) and Colang 2.0 (`flows.co`) flow definitions for `zscaler aiguard moderation on input` and `zscaler aiguard moderation on output`
- **Dual policy modes** — Supports `resolve-and-execute-policy` (automatic, default) and `execute-policy` (explicit policy ID via `AIGUARD_POLICY_ID` env var or action parameter)
- **Fail-closed semantics** — API failures or unexpected results default to BLOCK
- **Rails exceptions** — When `enable_rails_exceptions: true` is set, blocked requests emit `ZscalerAiguardInputRailException` / `ZscalerAiguardOutputRailException` with severity, policy name, blocking detectors, and transaction ID
- **Documentation** — `docs/user-guides/community/zscaler-aiguard.md` with setup instructions for Colang v1 and v2
- **Comprehensive tests** — 20 tests (integration + unit) covering allow/block, error handling, policy ID routing, exception messages, and edge cases
- **Changelog** — Entry added under `[Unreleased]`

### Files added/modified

| File | Description |
|------|-------------|
| `nemoguardrails/library/zscaler_aiguard/__init__.py` | Package init |
| `nemoguardrails/library/zscaler_aiguard/actions.py` | System action with SDK integration |
| `nemoguardrails/library/zscaler_aiguard/flows.co` | Colang 2.0 flows |
| `nemoguardrails/library/zscaler_aiguard/flows.v1.co` | Colang 1.0 flows |
| `tests/test_zscaler_aiguard.py` | 20 tests (8 integration, 12 unit) |
| `tests/test_configs/zscaler_aiguard/config.yml` | Test config with exceptions enabled |
| `tests/test_configs/zscaler_aiguard_exceptions/config.yml` | Test config for exception tests |
| `docs/user-guides/community/zscaler-aiguard.md` | User guide documentation |
| `CHANGELOG.md` | Unreleased entry |

### Environment variables

| Variable | Required | Description |
|----------|----------|-------------|
| `AIGUARD_API_KEY` | Yes | Zscaler AI Guard API key (Bearer token) |
| `AIGUARD_CLOUD` | Yes | Cloud region (`us1`, `us2`, `eu1`, `eu2`) |
| `AIGUARD_POLICY_ID` | No | Specific policy ID (default: auto-resolved) |

### Dependencies

Requires `zscaler-sdk-python` (listed as optional — users install it separately).

## Test plan

- [x] All 20 plugin tests pass (`poetry run pytest tests/test_zscaler_aiguard.py`)
- [x] Integration tests verify input/output allow, block, DETECT, and API error scenarios
- [x] Unit tests verify action logic, policy ID routing (param, env var, override, invalid), message building, empty text, fail-closed
- [x] Exception tests verify `ZscalerAiguardInputRailException` / `ZscalerAiguardOutputRailException` contain severity, policy, and detectors
- [x] Tested interactively with `nemoguardrails chat` against live AI Guard API
- [x] Code formatted with Black
- [x] Changelog updated

Signed-off-by: William Guilherme <wguilherme@securitygeek.io>

Made with [Cursor](https://cursor.com)

## Release Notes

* **New Features**
  * Added Zscaler AI Guard integration enabling input/output content moderation with detection for credentials, PII, toxicity, prompt injection, malicious URLs, and policy violations.

* **Documentation**
  * Comprehensive user guide covering setup, configuration, environment variables, and integration examples.

* **Tests**
  * Test suite and sample configurations for the integration.

### CODE REVIEW

**Comment by willguibr** [comment]
## CI Note: Python 3.13 / Ubuntu failure

The `pr-tests-matrix (Ubuntu, 3.13)` failure is **not related to this PR**. The test that crashes is `tests/runnable_rails/test_history.py::test_message_history_with_rails` with `Illegal instruction (core dumped)` (SIGILL, exit code 132).

This is a known issue affecting **all current PRs** since the dependency update in #1617 was merged to `develop` on March 6, 2026. For example:
- PR #1702 — fails on Python 3.13 (same crash)
- PR #1708 — fails on Python 3.13 (same crash)

PRs merged before March 6 (e.g., #1685, #1687) passed Python 3.13 because they didn't include the updated `poetry.lock`.

This is tracked in **#1659** which suggests setting `NPY_DISABLE_CPU_FEATURES` in the CI workflow as a fix.

All other CI checks pass:
- ✅ Python 3.10
- ✅ Python 3.11
- ✅ Python 3.12
- ✅ Lint Code (after formatting fix)
- ✅ build-docs

File: CHANGELOG.md
Comment (Pouyanpi) [inline CHANGELOG.md:1]: please revert this. The changelog gets autogenerated.

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        if result is None:
+            log.warning("AI Guard returned None — blocking by default")
+            return {
+                "action": "BLOCK",
+                "severity": "UNKNOWN",
+                "detectors": {},
+                "blocking_detectors": [],
+                "message": _build_block_message(direction, "UNKNOWN", "unknown", []),
+            }
+
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py:183]: When `result is None`, the return dict is missing `policy_name` and `transaction_id` keys.

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        log.error("AI Guard scan failed: %s — %s", type(e).__name__, e)
+        message = _build_block_message(direction, "UNKNOWN", "unknown", [])
+        return {
+            "action": "BLOCK",
+            "severity": "UNKNOWN",
+            "detectors": {},
+            "blocking_detectors": [],
+            "error": str(e),
+            "message": message,
+        }
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py:253]: Missing `policy_name` and `transaction_id` in the exception/error return dict

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            raise ImportError(
+                "zscaler-sdk-python is required for the Zscaler AI Guard integration. "
+                "Install it with: pip install zscaler-sdk-python"
+            )
+
+        if not os.environ.get("AIGUARD_API_KEY"):
+            raise EnvironmentError(
+                "AIGUARD_API_KEY environment variable is required for the Zscaler AI Guard integration."
+            )
+
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py:65]: The code validates that `AIGUARD_API_KEY` is set, but never passes it to `LegacyZGuardClientHelper`. So it is checked but never used. Why is that?

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+import logging
+import os
+from typing import Any, Optional
+
+from nemoguardrails.actions import action
+
+log = logging.getLogger(__name__)
+
+_sdk_client = None
+_sdk_client_cloud = None
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py:43]: Module level singleton is not thread-safe. The global `_sdk_client` / `_sdk_client_cloud` is read and written without any lock. Since `call_zscaler_aiguard_api` is async and use `asyncio.to_thread`, concurrent requests could race on client initialization. In some cases it leads to double initialization but is worth to note. do you see any issues?

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+        action_val = str(_get_attr(result, "action", "BLOCK")).upper()
+        severity = _get_attr(result, "severity", "unknown")
+        policy_name = _get_attr(result, "policy_name") or _get_attr(result, "policyName", "unknown")
+        transaction_id = _get_attr(result, "transaction_id") or _get_attr(result, "transactionId")
+
+        detector_responses = _get_attr(result, "detector_responses") or _get_attr(result, "detectorResponses") or {}
+        detectors = {}
+        blocking_detectors = []
+        for name, det in detector_responses.items() if isinstance(detector_responses, dict) else []:
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py]: it silently skips non dict responses. at minimum, a log warning for the non dict case would help debuggability.

File: tests/test_zscaler_aiguard.py [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+        config,
+        llm_completions=["I don't know the answer to that."],
+    )
+
+    async def mock_action(text=None, direction="IN", **kwargs):
+        return _block_result(
+            direction="IN",
+            blocking_detectors=["pii", "secrets"],
+            message="Zscaler AI Guard blocked the user prompt. Severity: CRITICAL. Policy: PolicyApp01. Detectors: pii, secrets. Transaction: txn-001.",
+        )
```
Comment (Pouyanpi) [inline tests/test_zscaler_aiguard.py:309]: This means the test is asserting on a value it injected itself, not on `_build_block_message()` output. It doesn't actually test that the action builds the message correctly e2e

File: nemoguardrails/library/zscaler_aiguard/actions.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        _sdk_client = LegacyZGuardClientHelper(cloud=cloud)
+        _sdk_client_cloud = cloud
+    return _sdk_client
+
+
+def _get_attr(obj, name, default=None):
+    """Access an attribute by name from either a dict or an object."""
+    if isinstance(obj, dict):
+        return obj.get(name, default)
+    return getattr(obj, name, default)
```
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/actions.py:75]: `_get_attr` handles dict or object but SDK response type is unclear:

it seems that this `_get_attr` helper and the dual snake_case and camelCase lookups (`policy_name` / `policyName`, `transaction_id` / `transactionId`, etc.) suggest uncertainty about what the SDK actually returns. The code defensively handles both dicts and objects with both naming conventions. This works but is fragile.  if the sdk changes its return type, the fallback chain might silently return `None` / `"unknown"` for fields that should have real values. a single integration test against the real SDK would clarify this, and the code could be tightened. any idea on how we can run the integration tests?

File: nemoguardrails/library/zscaler_aiguard/flows.co
Comment (Pouyanpi) [inline nemoguardrails/library/zscaler_aiguard/flows.co:1]: have you tested using colang 2.0?

File: tests/test_configs/zscaler_aiguard_exceptions/config.yml
Comment (Pouyanpi) [inline tests/test_configs/zscaler_aiguard_exceptions/config.yml:1]: nit:  could be better to enable it programmatically instead of creating a new config file.

File: tests/test_configs/zscaler_aiguard/config.yml
Comment (Pouyanpi) [inline tests/test_configs/zscaler_aiguard/config.yml:1]: prefer passing the config programmatically using `RailsConfig.from_content` by setting it to `yaml_content` similar to `test_trend_ai_guard.py`.

File: tests/test_zscaler_aiguard.py
Comment (Pouyanpi) [inline tests/test_zscaler_aiguard.py:1]: missing test:

- there are tests for input exception (`ZscalerAiguardInputRailException`) but no test verifying `ZscalerAiguardOutputRailException` fires correctly with `enable_rails_exceptions: true`.

**Review by Pouyanpi — COMMENTED** [review]
Thank you @willguibr . Please address the remaining comments then we're good to merge. Also make sure that all your commits are gpg signed according to contributing guides. 

@miyoungc could you review the doc changes? thanks!

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 3 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py:65]: AIGUARD_API_KEY checked but not passed

File: nemoguardrails/library/zscaler_aiguard/flows.co
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/flows.co:1]: We use Colang 1.0, same as all other guard integrations in this repo (Trend AI Guard, etc.).

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
+# Zscaler AI Guard Integration
+
+The Zscaler AI Guard guardrail uses the [Zscaler AI Guard](https://help.zscaler.com/ai-guard) DAS API to scan prompts and LLM responses for security threats, including:
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Remove undefined "DAS" acronym — use plain "API":

```suggestion
The Zscaler AI Guard guardrail uses the [Zscaler AI Guard](https://help.zscaler.com/ai-guard) API to scan prompts and LLM responses for security threats, including:
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+- **Credentials and secrets** - API keys, tokens, passwords, and cloud credentials
+- **PII (Personally Identifiable Information)** - Names, emails, phone numbers, SSNs, and other personal data
+- **Toxicity** - Violent, abusive, hateful, or harmful content
+- **Prompt injection** - Adversarial prompts designed to manipulate AI behavior or bypass security controls
+- **Malicious URLs** - Known malicious URLs and domains
+- **Policy violations** - Custom content policy violations configured in the AI Guard console
+
+The integration uses the [zscaler-sdk-python](https://pypi.org/project/zscaler-sdk-python/) SDK and implements **fail-closed** semantics: if the API call fails or returns an unexpected result, the content is blocked by default.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Fix passive voice (style guide: use active voice whenever possible):

```suggestion
The integration uses the [zscaler-sdk-python](https://pypi.org/project/zscaler-sdk-python/) SDK and implements **fail-closed** semantics: if the API call fails or returns an unexpected result, the integration blocks the content by default.
```

**Review by miyoungc — COMMENTED** [review]
## NVIDIA Style Guide Review

Applied the NVIDIA documentation style guide review. The following suggestions fix 9 issues:

| # | Rule | Issue |
|---|------|-------|
| 1 | Unexplained acronym | "DAS API" — acronym not defined |
| 2 | Passive voice | "the content is blocked" → active voice |
| 3 | UI navigation formatting | Plain text → **bold** with `>` separators |
| 4 | Lead-in clarity | "Optionally, to use..." → imperative lead-in |
| 5 | Nonessential which-clause | Missing comma before "which" |
| 6 | Sentence >30 words + passive | Split long sentence, active voice |
| 7 | Latinism "etc." | Replace with specific examples |
| 8 | Passive voice | "Detectors are configured" → imperative |
| 9 | Latinism "via" | "via" → "through" |

Please accept the inline suggestions below to apply all fixes.

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+The following environment variables are required:
+
+- `AIGUARD_API_KEY`: Zscaler AI Guard API key (Bearer token). Obtain from AI Guard Console → Private AI Apps → Applications → API Keys.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Bold UI elements and use `>` for navigation paths (style guide: UI elements in bold, navigation with angle brackets):

```suggestion
- `AIGUARD_API_KEY`: Zscaler AI Guard API key (Bearer token). To obtain this key, go to **AI Guard Console** > **Private AI Apps** > **Applications** > **API Keys**.
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+- `AIGUARD_CLOUD`: Cloud region. Options: `us1` (default), `us2`, `eu1`, `eu2`.
+
+Optionally, to use a specific policy instead of automatic resolution:
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Improve lead-in sentence clarity:

```suggestion
To use a specific policy instead of automatic resolution, set the following optional variable:
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+- `AIGUARD_POLICY_ID`: Integer policy ID. When set, the integration calls `execute-policy` with the given ID instead of `resolve-and-execute-policy`.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Use "specified" for consistency:

```suggestion
- `AIGUARD_POLICY_ID`: Integer policy ID. When set, the integration calls `execute-policy` with the specified ID instead of `resolve-and-execute-policy`.
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+flow output rails $output_text
+    zscaler aiguard moderation on output
+```
+
+## How It Works
+
+1. **Input scanning**: Before user prompts reach the LLM, the `zscaler aiguard moderation on input` flow sends the prompt to the AI Guard API with `direction="IN"`.
+2. **Output scanning**: After the LLM generates a response, the `zscaler aiguard moderation on output` flow sends the response to the AI Guard API with `direction="OUT"`.
+3. **Policy selection**: By default, the integration calls `resolve-and-execute-policy` which automatically selects the appropriate policy. If `AIGUARD_POLICY_ID` is set, it instead calls `execute-policy` with the specified policy ID.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Add comma before nonessential "which" clause; use active voice:

```suggestion
3. **Policy selection**: By default, the integration calls `resolve-and-execute-policy`, which automatically selects the appropriate policy. If `AIGUARD_POLICY_ID` is set, the integration calls `execute-policy` with the specified policy ID instead.
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+4. **Verdict handling**: If the API returns an `action` of `BLOCK`, the flow aborts and the bot refuses to respond. If the API returns `ALLOW` or `DETECT`, the content passes through normally.
+5. **Rails exceptions**: When `enable_rails_exceptions: true` is set at the top level of the config, blocked requests emit a `ZscalerAiguardInputRailException` or `ZscalerAiguardOutputRailException` with a detailed message containing the severity, policy name, blocking detectors, and transaction ID.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Split sentence exceeding 30 words and fix passive voice:

```suggestion
5. **Rails exceptions**: Setting `enable_rails_exceptions: true` at the top level of the config causes blocked requests to emit a `ZscalerAiguardInputRailException` or `ZscalerAiguardOutputRailException`. The exception message contains the severity, policy name, blocking detectors, and transaction ID.
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+6. **Fail-closed**: If the API call fails (network error, timeout, etc.), the action returns `action: BLOCK` by default to prevent potentially unsafe content from passing through.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Replace Latinism "etc." with specific examples:

```suggestion
6. **Fail-closed**: If the API call fails (network error, timeout, or authentication failure), the action returns `action: BLOCK` by default to prevent potentially unsafe content from passing through.
```

File: docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+- `toxicity` - Toxic or harmful language
+- `pii` - Personally identifiable information
+- `personalData` - Personal data patterns
+- `piiDeepscan` - Deep PII scanning
+- `secrets` - Credentials, API keys, tokens
+- `promptInjection` - Prompt injection attempts
+- `maliciousUrl` - Malicious URL detection
+
+Detectors are configured in the [Zscaler AI Guard console](https://help.zscaler.com/aiguard), not in the NeMo Guardrails configuration.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/community/zscaler-aiguard.md]: Fix passive voice to imperative:

```suggestion
Configure detectors in the [Zscaler AI Guard console](https://help.zscaler.com/aiguard), not in the NeMo Guardrails configuration.
```

File: docs/configure-rails/guardrail-catalog/third-party.md [hunk 17]
```diff
@@ -284,6 +284,26 @@ rails:
 
 For more details, check out the [Cisco AI Defense Integration](community/ai-defense.md) page.
 
+## Zscaler AI Guard
+
+The NeMo Guardrails library supports using [Zscaler AI Guard](https://www.zscaler.com/platform/ai-guard) for input and output content moderation via the `zscaler-sdk-python` SDK. It detects credentials, PII, toxicity, prompt injection, malicious URLs, and policy violations.
```
Comment (miyoungc) [inline docs/configure-rails/guardrail-catalog/third-party.md]: Replace Latinism "via" with "through":

```suggestion
The NeMo Guardrails library supports using [Zscaler AI Guard](https://www.zscaler.com/platform/ai-guard) for input and output content moderation through the `zscaler-sdk-python` SDK. It detects credentials, PII, toxicity, prompt injection, malicious URLs, and policy violations.
```

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 1 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py:183]: Fixed in 122eaa5. Both policy_name and transaction_id are now included in the result is None return path (and the exception path) for a consistent return schema across all code paths.

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 2 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py:253]: Fixed in 122eaa5. Added "policy_name": "unknown" and "transaction_id": None to the exception/error return dict.

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 4 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py:43]: Acknowledged. The worst case is double initialization — two identical LegacyZGuardClientHelper instances created — which is benign since the client is stateless (it just holds config: API key, cloud URL, headers). The reference assignment is atomic under CPython's GIL, so there's no corruption risk. After the race, _sdk_client always points to a valid client and all subsequent calls reuse it. Happy to add a threading.Lock if you prefer, but the impact is a single redundant instantiation with no functional side effects.

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 5 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py]: fixed in 122eaa5. Added an explicit log.warning when detector_responses is not a dict before falling back to an empty dict.

File: tests/test_zscaler_aiguard.py (hunk 6 above)
Comment (willguibr) [reply tests/test_zscaler_aiguard.py:309]: This test is intentionally not testing _build_block_message() — it tests the Colang flow integration: that when the action returns a BLOCK with a message, the ZscalerAiguardInputRailException correctly propagates that message through the flow to the client. _build_block_message() is covered separately by the unit test test_zscaler_aiguard_build_block_message and end-to-end through test_zscaler_aiguard_action_scan_block which uses mocked SDK responses.

File: nemoguardrails/library/zscaler_aiguard/actions.py (hunk 7 above)
Comment (willguibr) [reply nemoguardrails/library/zscaler_aiguard/actions.py:75]: The _get_attr helper handles both dict and object access because the SDK returns ZscalerObject instances (not dicts). The SDK normalizes API JSON keys to snake_case internally, so the response always uses policy_name, transaction_id, detector_responses etc. The camelCase fallbacks are redundant — I can remove them to simplify the code. Regarding integration tests, they require a live Zscaler AI Guard tenant with valid AIGUARD_API_KEY credentials which aren't available in CI. Happy to add one if there's a credential injection mechanism in the NeMo Guardrails CI pipeline.

File: tests/test_configs/zscaler_aiguard/config.yml
Comment (willguibr) [reply tests/test_configs/zscaler_aiguard/config.yml:1]: We already use RailsConfig.from_content for 3 inline configs at the top of the test file (input_rail_config, output_rail_config, both_rails_config). The file-based configs are kept intentionally to also cover the from_path loading path. Happy to consolidate to inline-only if you prefer.

File: tests/test_configs/zscaler_aiguard_exceptions/config.yml
Comment (willguibr) [reply tests/test_configs/zscaler_aiguard_exceptions/config.yml:1]: Ack — will convert this to an inline RailsConfig.from_content with enable_rails_exceptions: true and remove the config file.

File: tests/test_zscaler_aiguard.py
Comment (willguibr) [reply tests/test_zscaler_aiguard.py:1]: Added test_zscaler_aiguard_output_exception which verifies ZscalerAiguardOutputRailException fires correctly with enable_rails_exceptions: true. See latest commit.

File: CHANGELOG.md
Comment (willguibr) [reply CHANGELOG.md:1]: Reverted in 122eaa5. The CHANGELOG.md is back to its upstream state.

File: nemoguardrails/library/zscaler_aiguard/flows.co
Comment (Pouyanpi) [reply nemoguardrails/library/zscaler_aiguard/flows.co:1]: flows.co enables users to use colang 2.0, can you simply run an example with colang 2.0 to see it works. You can check `examples/bots/abc_v2` or `examples/configs/nemoguards_v2`
